### PR TITLE
Hide publish button when video publishing is unavailable

### LINKED ISSUE #3807: Hide publish button when video quota is 0
**Describe the current behavior**
The publish button in the right corner is always shown.

**Describe the expected behavior**
A) When a user is not logged in it should be hidden if the video quota is 0 for new users.
B) When a user is logged in it should be hidden if the video quota is 0 for the current user.

The button is one of few buttons and therefore takes a lot of user focus. And when clicked on the button (when video quota is 0) it looks like something isn't working properly.

### PR DESCRIPTION
**Pull request by kontrollanten**
## Description
Hides publish button, top right corner, when uploads are disabled.

## Related issues
Closes #3807 

## Has this been tested?

- [x] 👍 yes, light tests as follows are enough

1) Set video quota for new users to unlimited.
1) Logout.
1) See the upload button.

1) Set video quota for new users to 0.
1) Logout.
1) Don't see the upload button.

1) Set the video quota for new users to 0.
1) Set the video quota for a particular user to unlimited.
1) Login as that user.
1) See the publish button.

## Screenshots

image

### CODE REVIEW

File: client/src/app/header/header.component.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 
 @Component({
   selector: 'my-header',
   templateUrl: './header.component.html',
   styleUrls: [ './header.component.scss' ]
 })
 
-export class HeaderComponent {}
+export class HeaderComponent {
+  @Input() showPublishButton: boolean
```
Comment (rigelk) [inline client/src/app/header/header.component.ts:10]:
```suggestion
  @Input() showPublishButton = true
```

File: client/src/app/app.component.ts [hunk 2]
```diff
@@ -115,6 +115,14 @@ export class AppComponent implements OnInit, AfterViewInit {
     return this.authService.isLoggedIn()
   }
 
+  isUploadEnabled () {
+    if (this.authService.isLoggedIn()) {
+      return this.authService.getUser().videoQuota !== 0
+    } else {
+      return this.serverConfig.user.videoQuota !== 0
+    }
```
Comment (rigelk) [inline client/src/app/app.component.ts:123]:
```suggestion
    return this.authService.isLoggedIn()
      ? this.authService.getUser().videoQuota !== 0
      : this.serverConfig.user.videoQuota !== 0
```

**Review by rigelk — CHANGES_REQUESTED** [review]

**Comment by kimsible** [comment]
Hmm hiding the publish is now in the user scenario @rigelk ?

https://github.com/Chocobozzz/PeerTube/pull/3357#issuecomment-733666215

**Comment by rigelk** [comment]
The user is logged in and doesn't have any quota, that's not the same situation.

**Comment by kontrollanten** [comment]
What's the status of this? Is there anything I can do?

**Comment by Chocobozzz** [comment]
Sorry for the late review/answer @kontrollanten

Hiding the upload button has already been discussed in https://github.com/Chocobozzz/PeerTube/pull/3357 so it's not a feature we want to implement.

What I can do is to add a `userInformationLoaded` plugin hook, and a `getServerConfig` helper on the client so admins can create/install a plugin that hides this upload button depending on their need.

**Comment by kimsible** [comment]
> Sorry for the late review/answer @kontrollanten
> 
> Hiding the upload button has already been discussed in #3357 so it's not a feature we would like to implement.
> 
> What I can do is to add a `userInformationLoaded` plugin hook, and a `getServerConfig` helper on the client so admins can create/install a plugin that hides this upload button depending on their need.

Thanks @Chocobozzz, sounds a very good compromise.

>> In my opinion the real problem is that publish button is confusing with the search box at left, we may click on the publish button to run a search.

> This is exactly the problem we're facing. But that's a separate issue then I guess.

Yep @kontrollanten it has been reported by some of our members too. If it can help we tried to hack it with a simple CSS override but you'll have a problem with the main container search and the auto-complete will not follow. So yes, in my opinion, the only way to resolve properly this issue (or center the search box ? many things could actually be proposed) is to edit the template angular source file.

**Comment by Chocobozzz** [comment]
> This is exactly the problem we're facing. But that's a separate issue then I guess.

A UX work on the header is planned

> What I can do is to add a userInformationLoaded plugin hook, and a getServerConfig helper on the client so admins can create/install a plugin that hides this upload button depending on their need.

Done in https://github.com/Chocobozzz/PeerTube/commit/7bc572c7531580600d02b4fc5f1b6cf54df139f0

**Pull request closed without merging**